cypress/e2e/editing/editing.spec.ts

```diff
@@ -55,14 +60,269 @@ describe("Editing", () => {
         cy.stopHomeserver(homeserver);
     });
 
-    it("should close the composer when clicking save after making a change and undoing it", () => {
-        cy.createRoom({ name: "Test room" }).as("roomId");
+    it("should render and interact with the message edit history dialog", () => {
+        cy.visit("/#/room/" + roomId);
+
+        // Send "Message"
+        sendEvent(roomId);
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            // Edit message
+            cy.get(".mx_EventTile_last").realHover();
+            cy.get(".mx_EventTile_last .mx_MessageActionBar_optionsButton", { timeout: 1000 })
+                .should("exist")
+                .realHover()
+                .get('.mx_EventTile_last [aria-label="Edit"]')
+                .click({ force: false });
+            cy.get(".mx_BasicMessageComposer_input").type("{selectAll}{del}Massage{enter}");
+        });
+
+        // Assert that the edit label is visible
+        cy.get(".mx_EventTile_edited").should("be.visible");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            // Assert that the message was edited
+            cy.contains(".mx_EventTile", "Massage")
+                .should("exist")
+                .within(() => {
+                    // Click to display the message edit history dialog
+                    cy.contains(".mx_EventTile_edited", "(edited)").click();
+                });
+        });
+
+        cy.get(".mx_Dialog").within(() => {
+            // Assert that the message edit history dialog is rendered
+            cy.get(".mx_MessageEditHistoryDialog").within(() => {
+                // Assert that CSS styles which cannot be detected with snapshots are applied as expected
+                cy.get("li").should("have.css", "clear", "both");
+                cy.get(".mx_EventTile")
+                    .should("have.css", "max-width", "100%")
+                    .should("have.css", "clear", "both")
+                    .should("have.css", "position", "relative")
+                    .should("have.css", "padding-block-start", "0px");
+                cy.get(".mx_EventTile .mx_MessageTimestamp")
+                    .should("have.css", "position", "absolute")
+                    .should("have.css", "inset-inline-start", "0px")
+                    .should("have.css", "text-align", "center");
+                cy.get(".mx_EventTile .mx_EventTile_content").should("have.css", "margin-inline-end", "0px");
+
+                // Assert that the date separator is rendered
+                cy.get("li:nth-child(1) .mx_DateSeparator").within(() => {
+                    cy.get("h2").should("have.text", "Today");
+                });
+
+                // Assert that the edited message is rendered
+                cy.get("li:nth-child(2) .mx_EventTile").within(() => {
+                    cy.get(".mx_EventTile_content").within(() => {
+                        cy.get(".mx_EventTile_body").should("have.text", "Meassage");
```

Comment:
Can you add a comment why it is „Meassage“  here?
Took me some moments to understand :hourglass:

Reply:
This commit https://github.com/matrix-org/matrix-react-sdk/pull/10406/commits/b930982d1a7a63996802e907c786f7fa87539853 added a comment to explain it.